docs/migration-guide.asciidoc

```diff
@@ -0,0 +1,311 @@
+[[migration-guide]]
+== Migration guide NEST -> v8.*
```

Comment:
The title you suggested is great. I also recommend the one below. It might not be so appealing though. Please use what fits better in your opinion.

```suggestion
== Migration guide: From NEST v7 to .NET Client v8
```